make.go

```diff
@@ -152,33 +165,34 @@ func (c *pChunker) start(ctx context.Context) {
 			return
 		default: // We weren't asked to stop and weren't interrupted, carry on
 		}
-		start, b, err := c.chunker.Next()
+		chunk, err := c.chunker.Next()
 		if err != nil {
 			c.err = err
 			return
 		}
 		c.stats.incProduced()
-		start += c.offset
-		if len(b) == 0 {
+		chunk.Start += c.offset
+		if len(chunk.Data) == 0 {
 			// TODO: If this worker reached the end of the stream and it's not the
 			// last one, we should probable stop all following workers. Meh, shouldn't
 			// be happening for large file or save significant CPU for small ones.
 			c.eof = true
 			return
 		}
-		// Calculate the chunk ID
-		id := sha512.Sum512_256(b)
-
 		// Store it in our bucket
-		chunk := IndexChunk{Start: start, Size: uint64(len(b)), ID: id, b: b}
-		c.results <- chunk
+		ci := NewChunkIndex(chunk)
+		c.results <- ci
 
 		// Check if the next worker already has this chunk, at which point we stop
 		// here and let the next continue
-		if c.next != nil && c.next.syncWith(chunk) {
+		if c.next != nil && c.next.syncWith(ci) {
 			return
 		}
 
+		if c.s != nil {
+			c.err = c.s.StoreChunk(chunk)
```

Comment:
The chunks produced during parallel chunking aren't confirmed yet and shouldn't be stored yet.

Reply:
I'll revert this in the next rev.